### PR TITLE
Make POSIX-compliant to increase portability and remove `bash` depend…

### LINKED ISSUE #100144: Make `resources/linux/bin/code.sh` POSIX-compliant to remove `bash` dependency
I'd like `resources/linux/bin/code.sh` to be POSIX-compliant to remove the dependency on `bash`.

### PR DESCRIPTION
**Pull request by nothingnesses**
…ency on Linux. Fix causes of `shellcheck` warnings.

This PR fixes #100144 by removing bashisms and replacing them with POSIX-compliant alternatives.

### CODE REVIEW

**Comment by VoidNoire** [comment]
Should the `VS Code` on line 33 be replaced with `@@PRODNAME@@`?

File: resources/linux/bin/code.sh [hunk 1]
```diff
@@ -1,28 +1,36 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
 #
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 
 # test that VSCode wasn't installed inside WSL
 if grep -qi Microsoft /proc/version && [ -z "$DONT_PROMPT_WSL_INSTALL" ]; then
 	echo "To use VS Code with the Windows Subsystem for Linux, please install VS Code in Windows and uninstall the Linux version in WSL. You can then use the '@@PRODNAME@@' command in a WSL terminal just as you would in a normal command prompt." 1>&2
```
Comment (VoidNoire) [inline resources/linux/bin/code.sh:8]:
```suggestion
	echo "To use @@PRODNAME@@ with the Windows Subsystem for Linux, please install @@PRODNAME@@ in Windows and uninstall the Linux version in WSL. You can then use the \`@@NAME@@\` command in a WSL terminal just as you would in a normal command prompt." 1>&2
```

File: resources/linux/bin/code.sh [hunk 2]
```diff
@@ -1,28 +1,36 @@
...
-		fi
+		case "$i" in
+			--user-data-dir | --user-data-dir=* | --file-write )
+				CAN_LAUNCH_AS_ROOT=1
+			;;
+		esac
 	done
 	if [ -z $CAN_LAUNCH_AS_ROOT ]; then
-		echo "You are trying to start vscode as a super user which is not recommended. If you really want to, you must specify an alternate user data directory using the --user-data-dir argument." 1>&2
+		echo "You are trying to start VS Code as a super user which isn't recommended. If this was intended, please specify an alternate user data directory using the \`--user-data-dir\` argument." 1>&2
```
Comment (VoidNoire) [inline resources/linux/bin/code.sh:33]:
```suggestion
		echo "You are trying to start @@PRODNAME@@ as a super user which isn't recommended. If this was intended, please specify an alternate user data directory using the \`--user-data-dir\` argument." 1>&2
```

**Review by nothingnesses — COMMENTED** [review]
Should the VS Code on lines 8 and 33 be replaced with @@PRODNAME@@?

**Comment by joaomoreno** [comment]
Does this fix an _actual_ issue?

**Comment by VoidNoire** [comment]
I'm unable to start VS Code on my system which doesn't have `bash`, and this fixes that.

**Comment by joaomoreno** [comment]
Can you give me an example of said system?

**Comment by VoidNoire** [comment]
Sure thing. There are certain Linux distributions that don't enforce the usage of, and don't come pre-packaged with `bash`. For example, there's the [minimal version](https://github.com/void-linux/void-packages/blob/master/srcpkgs/base-minimal/template) of the Void Linux distribution which opts for `dash` and [Kiss Linux](https://k1ss.org/software#1.2) which opts for `busybox` instead. On these systems, `bash` would arguably be considered redundant and could (in the case of Kiss where all packages must be compiled by the user) cause other software that depend on it an unnecessarily longer amount of time to build (when considering that `bash` also has to be built even though all it might be needed for is to run a start up script for VS Code). As such, reducing dependencies would be beneficial for the end user.

**Comment by joaomoreno** [comment]
I'm actually surprised VS Code runs in said systems. 👌

File: resources/linux/bin/code.sh (hunk 1 above)
Comment (joaomoreno) [reply resources/linux/bin/code.sh:8]: cc @aeschli

File: resources/linux/bin/code.sh (hunk 2 above)
Comment (joaomoreno) [reply resources/linux/bin/code.sh]: cc @bpasero

**Comment by joaomoreno** [comment]
@aeschli @bpasero Please review whether your areas of the script still work as designed.

**Review by joaomoreno — APPROVED** [review]

**Review by bpasero — APPROVED** [review]

**Review by aeschli — APPROVED** [review]

**Pull request merged**